### PR TITLE
Feature/fix fetch calls

### LINKED ISSUE #11: Add a loader to frontend that loads until the data arrives

### LINKED ISSUE #12: Add FETCH error handling (proper UI)

### LINKED ISSUE #13: Refactor FE to reflect FETCH endpoints

### PR DESCRIPTION
**Pull request by hvohr**
What I did: Refactored the front end to reflect the new backend GET endpoints. No savedData information (POST or DELETE) was changed. I also added extensive error management for fetch failure and loading tickets for UI.

Did this break anything?

- [x] No
- [ ]  Yes

Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ]  Bug fix (non-breaking change which fixes an issue)
- [x]  Refactor(DRY-ing up/ reorganizing code, etc.)
- [ ]  Super small fix (Corrected a typo, removed a comment, etc.)
- [ ]  Skip all the other stuff and briefly explain the fix.

Checklist:

- [ ]  If this code needs to be tested, all tests are passing
- [x]  The code runs locally
- [ ]  There are comments where clarification/ organization is needed
- [x]  The code is DRY. If it's not, I tried my best
- [x]  I reviewed my code before pushing

What's next? Fixing the POST and DELETE endpoints so they reflect the backend as well. The savedData file is still present, only the data file was deleted.

### CODE REVIEW

File: src/components/SelectTaskView/SelectTaskView.js [hunk 1]
```diff
@@ -21,153 +22,163 @@ const SelectTaskView = () => {
   const [saveResponse, setSaveResponse] = useState('');
   const [saveIcon, setSaveIcon] = useState(savePurpleIcon);
 
+  useEffect(() => {
+    fetchAllTasks().then(
+      data => setTasks(data)
+    )
```
Comment (taranveersingh93) [inline src/components/SelectTaskView/SelectTaskView.js:30]: hey! Given that you're throwing a new Error in the api calls at a not Ok response, I think you'd need a catch block here.

File: src/components/SelectTaskView/SelectTaskView.js [hunk 2]
```diff
@@ -13,6 +13,7 @@ const SelectTaskView = () => {
   const { category } = useParams();
 
   const [currentTasks, setCurrentTasks] = useState([]);
+  const [tasks, setTasks] = useState('')
```
Comment (taranveersingh93) [inline src/components/SelectTaskView/SelectTaskView.js]: should tasks be initialised as an empty array?

Comment (taranveersingh93) [reply src/components/SelectTaskView/SelectTaskView.js]: If you change this, you'd probably also need to change the conditional in getUnseenTasks

**Review by taranveersingh93 — CHANGES_REQUESTED** [review]
This was a tricky refactor. Really nice work!
I console logged unseen tasks and checked that they were being affected by refresh as expected (and it was great). 
Requested a couple changes.
I'm making a ticket for having a loader because when you load the page, you'd notice the error message for half a second (probably because there's no tasksToShow).

**Review by taranveersingh93 — APPROVED** [review]
Solid! thanks!

**Review by CaliHam — APPROVED** [review]
Sweet looks great!!

**Review by SethBurns — APPROVED** [review]
Looks awesome! Just gotta bugfix the mentalCare thing.

Nice work!

**Pull request merged**